### DISCUSSION TITLE
When converting pdf to html, how can I enable inputs in html?

### DISCUSSION THREAD

**Discussion by ashishelskiy** [pymupdf/PyMuPDF#1386] [Q&A]

**Comment by ashishelskiy** [comment]
[2.pdf](https://github.com/pymupdf/PyMuPDF/files/7512134/2.pdf)

**Comment by ashishelskiy** [comment]
![Screenshot from 2021-11-10 14-08-15](https://user-images.githubusercontent.com/8178216/141102689-9f150c0a-c738-40f3-9fae-9af05e9b7a04.png)

**Comment by ashishelskiy** [comment]
The screenshot shows the result in html format. There are no fields.

**Comment by ashishelskiy** [comment]
I did this:

doc = fitz.open ('2.pdf')
with open ('2.html', 'w') as f:
     for current_page in range (len (doc)):
         page = doc.loadPage (current_page)
         page_text = page.getText ("html")
         f.write (page_text)

Looks like I still need to apply this:

for field in page.widgets ():
     # do something with 'field'

But it displays all fields separately from the rest of the text.

**Comment by JorjMcKie** [comment]
MuPDF does not support annotations or fields to be preserved when converting to other types than PDF. Filled out field values seem to be converted okay, too.
But not as inputtable html fields.
